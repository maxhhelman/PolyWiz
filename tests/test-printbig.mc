/*
 * Test for linking external C functions to LLVM-generated code
 *
 * printbig is defined as an external function, much like printf
 * The C compiler generates printbig.o
 * The LLVM compiler, llc, translates the .ll to an assembly .s file
 * The C compiler assembles the .s file and links the .o file to generate
 * an executable
 */

int main()
{
  printbig(72); /* H */
  printbig(69); /* E */
  printbig(76); /* L */
  printbig(76); /* L */
  printbig(79); /* O */
  printbig(32); /*   */
  printbig(87); /* W */
  printbig(79); /* O */
  printbig(82); /* R */
  printbig(76); /* L */
  printbig(68); /* D */
  return 0;
}
