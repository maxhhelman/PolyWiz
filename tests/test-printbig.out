  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
        XX      
        XX      
        XX      
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
                
                
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
  XX    XX    XX
  XX    XX    XX
  XX    XX    XX
  XX          XX
                
                
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
  XX            
  XX            
  XX            
  XX            
                
                
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
  XX            
  XX            
  XX            
  XX            
                
    XXXXXXXXXX  
  XXXXXXXXXXXXXX
  XX          XX
  XX          XX
  XX          XX
  XXXXXXXXXXXXXX
    XXXXXXXXXX  
                
                
                
                
                
                
                
                
                
      XXXXXXXXXX
  XXXXXXXXXXXXXX
    XXXXXX      
      XXXXXX    
    XXXXXX      
  XXXXXXXXXXXXXX
      XXXXXXXXXX
                
    XXXXXXXXXX  
  XXXXXXXXXXXXXX
  XX          XX
  XX          XX
  XX          XX
  XXXXXXXXXXXXXX
    XXXXXXXXXX  
                
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
      XX      XX
    XXXX      XX
  XXXXXXXX    XX
  XXXX  XXXXXXXX
  XX    XXXXXX  
                
                
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
  XX            
  XX            
  XX            
  XX            
                
  XXXXXXXXXXXXXX
  XXXXXXXXXXXXXX
  XX          XX
  XX          XX
  XXXX      XXXX
    XXXXXXXXXX  
      XXXXXX    
                
